packages/astro/src/runtime/server/render/component.ts

```diff
@@ -478,7 +478,11 @@ export async function renderComponent(
 		return renderAstroComponent(result, displayName, Component, props, slots);
 	}
 
-	return await renderFrameworkComponent(result, displayName, Component, props, slots);
+	return await renderFrameworkComponent(result, displayName, Component, props, slots)
+		.catch(e => {
+			if (result.cancelled) return { render() {} };
+			throw e;
+		});
```

Comment:
The fix is here, the rest is clerical.

Reply:
Should we handle this catch for the entire code in `renderComponent` instead? Other components may error asynchronously too.

One of the caveats I figured with this technique is that we now have to ensure every async render APIs used in the template needs this special handling, but perhaps it's not too bad, we don't have many async APIs.

Reply:
> Should we handle this catch for the entire code in renderComponent instead?

Yeah, that's a good idea.

> One of the caveats I figured with this technique is that we now have to ensure every async render APIs used in the template needs this special handling

That isn't sufficient either - there could be offshoot promises created by the user or a library. The blanket fix would be handling unhandled rejections similar to the dev mode. The problem there is listening for them doesn't seem to have a standard way across runtimes - it will be per adapter. I'll open a PR for node tomorrow.

Reply:
Yeah that's true. I think for those cases, we should recommend not passing in promises instead. I think it's rare for someone to do that, so I'm not too concerned if it happens.

I'm not too sure yet if adapters should be injecting the unhandledrejections by default though. It can happen anywhere in the code (e.g. third-party libraries), so it should be up to users to handle it themselves.

Reply:
That's true, and we should prevent crashes in those situation - nothing more.

Reply:
> Should we handle this catch for the entire code in renderComponent instead?

Added cancellation handling for all async work in `renderComponent()`.